### PR TITLE
discovery: use token bucket based rate limiting to throttle gossip

### PR DESCRIPTION
**Pull request by wpaulino**
The recently added gossip throttling was shown to be too aggressive, especially with our auto channel enable/disable signaling. We switch to a token bucket based system instead as it's based on time, rather than a block height which isn't constantly updated at a given rate.

Along the way, we also revert the commits that exposed a config flag that was only necessary for our integration tests with the previous gossip throttling logic.

### CODE REVIEW

File: discovery/gossiper.go [hunk 1]
```diff
@@ -1950,16 +1974,28 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
...
+				rateLimiters, ok := d.chanUpdateRateLimiter[shortChanID]
+				if !ok {
+					r := rate.Every(d.cfg.ChannelUpdateInterval)
+					b := d.cfg.MaxChannelUpdateBurst
+					rateLimiters = [2]*rate.Limiter{
+						rate.NewLimiter(r, b),
+						rate.NewLimiter(r, b),
+					}
+					d.chanUpdateRateLimiter[shortChanID] = rateLimiters
+				}
```
Comment (Roasbeef) [inline discovery/gossiper.go:1992]: Do we still need to hold the mutex at this point? AFAICT, we have a pointer here, so we can release and then continue to make modifications to the underlying rate limiter instances.

File: discovery/gossiper.go [hunk 2]
```diff
@@ -335,7 +359,7 @@ func New(cfg Config, selfKey *btcec.PublicKey) *AuthenticatedGossiper {
 		prematureChannelUpdates: make(map[uint64][]*networkMsg),
 		channelMtx:              multimutex.NewMutex(),
 		recentRejects:           make(map[uint64]struct{}),
-		heightForLastChanUpdate: make(map[uint64][2]uint32),
+		chanUpdateRateLimiter:   make(map[uint64][2]*rate.Limiter),
```
Comment (Roasbeef) [inline discovery/gossiper.go:362]: Commenting here as can't below: so we're confident the keep-alive code isn't at fault, right?

File: discovery/gossiper.go (hunk 1 above)
Comment (wpaulino) [reply discovery/gossiper.go:1992]: We don't, the rate limiter is safe for concurrent access. Fixed.

File: discovery/gossiper.go (hunk 2 above)
Comment (wpaulino) [reply discovery/gossiper.go:362]: Yeah, so far the reports have been regarding updates that actually change something (e.g., disabled bit). Keep alive updates can't be done by the user at will within lnd, they're only sent every 24H.

File: discovery/gossiper.go [hunk 3]
```diff
@@ -22,6 +22,19 @@ import (
...
+const (
+	// DefaultMaxChannelUpdateBurst is the default maximum number of updates
+	// for a specific channel and direction that we'll accept over an
+	// interval.
+	DefaultMaxChannelUpdateBurst = 10
+
+	// DefaultChannelUpdateInterval is the default interval we'll use to
+	// determine how often we should allow a new update for a specific
+	// channel and direction.
+	DefaultChannelUpdateInterval = time.Minute
```
Comment (wpaulino) [inline discovery/gossiper.go:37]: These values were somewhat arbitrarily chosen in case reviewers have any thoughts/suggestions.

**Review by Roasbeef — APPROVED** [review]
LGTM 🥃

**Review by cfromknecht — APPROVED** [review]
LGTM 🎉 default parameters seem reasonable, we can tune further if anything pops up during the rc phase.

**Pull request merged**